cJSON.c

```diff
@@ -2082,6 +2082,165 @@ CJSON_PUBLIC(cJSON*) cJSON_AddArrayToObject(cJSON * const object, const char * c
     return NULL;
 }
 
+CJSON_PUBLIC(cJSON*) cJSON_AddNullToArray(cJSON *array)
+{
+    cJSON *null_item = cJSON_CreateNull();
+    if (array == NULL)
+    {
+        cJSON_Delete(null_item);
+        return NULL;
+    }
```

Comment:
The **add_item_to_array** function has already judged whether the **array** object is empty. There is no need to judge here.
```C
static cJSON_bool add_item_to_array(cJSON *array, cJSON *item)
{
    cJSON *child = NULL;

    if ((item == NULL) || (array == NULL))
    {
        return false;
    }

    child = array->child;

    if (child == NULL)
    {
        /* list is empty, start new one */
        array->child = item;
    }
    else
    {
        /* append to the end */
        while (child->next)
        {
            child = child->next;
        }
        suffix_object(child, item);
    }

    return true;
}
```